lib/evmone/instructions.hpp

```diff
@@ -400,25 +514,30 @@ inline void caller(StackTop stack, ExecutionState& state) noexcept
 
 inline void callvalue(StackTop stack, ExecutionState& state) noexcept
 {
-    stack.push(intx::be::load<uint256>(state.msg->value));
+    if (state.scallvalue) stack.push({intx::be::load<uint256>(state.msg->value), state.scallvalue});
```

Comment:
Why this branch?

Reply:
if scallvalue is unset, we are at depth 0 and the callvalue is statically know to be the one originating from the transaction, hence we push a pure value. otherwise we are in a nested call where we set scallvalue to be a symbolic value, in which case we should use that... i can add a comment in code